### PR TITLE
Added PointCloudXYZHSVtoXYZRGB function

### PR DESCRIPTION
**Pull request by gbernalr**
I don't know why this function had not been added before.
PointXYZRGBtoXYZHSV is there but PointCloudXYZHSVtoXYZRGB is not so i created one.
Based in PointCloudXYZRGBtoXYZHSV function.

### CODE REVIEW

**Comment by chuismiguel** [comment]
I would love this pull request to be checked and accepted as it is a funcionality that I would like to use in my 3D Vision Robot with RPI4B project. Thank you

File: common/include/pcl/point_types_conversion.h [hunk 1]
```diff
@@ -369,6 +369,25 @@ namespace pcl
...
+    out.height  = in.height;
+    for (const auto &point : in.points)
+    {
+      PointXYZRGB p;
+      PointXYZHSVtoXYZRGB (point, p);
+      out.push_back (p);
+    }
+  }
+
+  /** \brief Convert a XYZHSV point cloud to a XYZRGB
```
Comment (mvieth) [inline common/include/pcl/point_types_conversion.h:387]: It seems you have mixed up the documentation. Please keep the doc for the existing function the same, and add the appropriate doc for the new function.
Side note: don't worry about the failing test, that unfortunately happens sometimes (rarely). We didn't find a good solution for that yet. If you push the fix for this, the check will run again and most likely pass

Comment (gbernalr) [reply common/include/pcl/point_types_conversion.h]: Hi,

Check my last commit and please tell me if everything is now correct.

Sorry for the inconvenience.
G

**Comment by chuismiguel** [comment]
> I don't know why this function had not been added before.
> PointXYZRGBtoXYZHSV is there but PointCloudXYZHSVtoXYZRGB is not so i created one.
> Based in PointCloudXYZRGBtoXYZHSV function.

Also, without any intention of harming, I wanted to slightly point you out a small gramatical error on this text. 'had not been' is incorrect. Means nothing. The correct expression would be 'has not been added'. Don't worry, live and learn from mistakes!

LM

**Review by mvieth — APPROVED** [review]
Thanks!

File: common/include/pcl/point_types_conversion.h [hunk 2]
```diff
@@ -364,6 +364,24 @@ namespace pcl
...
+    * \param[in] in the input XYZHSV point cloud
+    * \param[out] out the output XYZRGB point cloud
+    */
+  inline void 
+  PointCloudXYZHSVtoXYZRGB (const PointCloud<PointXYZHSV>& in,
+                            PointCloud<PointXYZRGB>&       out)
+  {
+    out.width   = in.width;
+    out.height  = in.height;
+    for (const auto &point : in.points)
```
Comment (larshg) [inline common/include/pcl/point_types_conversion.h:378]: Can we avoid using the .points field?

Comment (mvieth) [reply common/include/pcl/point_types_conversion.h:378]: Simply removing that and only writing `in` should work, right?

Comment (gbernalr) [reply common/include/pcl/point_types_conversion.h:378]: The .points field is used in all other tranformation functions, i thought it was correct to use it.

Comment (gbernalr) [reply common/include/pcl/point_types_conversion.h]: I made it only removing the .points file from all in variables. You were right mvieth.

**Review by mvieth — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Pull request merged**